core/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -571,7 +572,7 @@ final SearchContext createContext(ShardSearchRequest request, @Nullable Engine.S
                     QueryParseContext queryParseContext = new QueryParseContext(indicesService.getIndicesQueryRegistry());
                     queryParseContext.reset(parser);
                     queryParseContext.parseFieldMatcher(parseFieldMatcher);
-                    parseSource(context, SearchSourceBuilder.fromXContent(parser, queryParseContext, aggParsers, suggesters));
+                    parseSource(context, SearchSourceBuilder.fromXContent(queryParseContext, aggParsers, suggesters));
```

Comment:
We really shouldn't be calling this queryParseContext anymore since its no longer just for queries but for the whole search request. Can we rename the class and variables to `searchParseContext` or something?